rust/noosphere-storage/src/implementation/native.rs

```diff
@@ -85,7 +85,12 @@ impl Store for NativeStore {
 
     /// Flushes pending writes if there are any
     async fn flush(&self) -> Result<()> {
-        self.db.flush_async().await?;
-        Ok(())
+        // `flush_async()` deadlocks until there's a new release
+        // of sled. In the mean time, wrap it in our own async.
+        // https://github.com/spacejam/sled/issues/1308
+        let db = self.db.clone();
+        tokio::spawn(async move { db.flush() })
```

Comment:
From a scheduling perspective, I'm not sure if it makes that much of a difference to wrap this with `tokio::spawn`. Maybe we should just do a blocking flush for now?